paz/processors/pose.py

```diff
@@ -127,3 +132,249 @@ def call(self, boxes2D):
             y_center = (z_center * v) / self.focal_length
             hands_center.append([x_center, y_center, z_center])
         return np.array(hands_center)
+
+
+class ComputeCameraParameter(Processor):
+    """Computes camera parameter given camera matrix
+    and scale normalization factor of translation.
+
+    # Arguments
+        camera_matrix: Array of shape `(3, 3)` camera matrix.
+        translation_scale_norm: Float, factor to change units.
+            EfficientPose internally works with meter and if the
+            dataset unit is mm for example, then this parameter
+            should be set to 1000.
+    """
+    def __init__(self, camera_matrix, translation_scale_norm):
+        self.camera_matrix = camera_matrix
+        self.translation_scale_norm = translation_scale_norm
+        super(ComputeCameraParameter, self).__init__()
+
+    def call(self, image_scale):
+        return np.array([self.camera_matrix[0, 0], self.camera_matrix[1, 1],
+                         self.camera_matrix[0, 2], self.camera_matrix[1, 2],
+                         self.translation_scale_norm, image_scale])
+
+
+class MatchPoses(Processor):
+    """Match prior boxes with ground truth poses.
+
+    # Arguments
+        prior_boxes: Numpy array of shape (num_boxes, 4).
+        iou: Float in [0, 1]. Intersection over union in which prior
+            boxes will be considered positive. A positive box is box
+            with a class different than `background`.
+    """
+    def __init__(self, prior_boxes, iou=.5):
+        self.prior_boxes = prior_boxes
+        self.iou = iou
+        super(MatchPoses, self).__init__()
+
+    def call(self, boxes, poses):
+        return match_poses(boxes, poses, self.prior_boxes, self.iou)
+
+
+class TransformRotation(Processor):
+    """Computes axis angle rotation vector from a rotation matrix.
+
+    # Arguments:
+        num_pose_dims: Int, number of dimensions of pose.
+
+    # Returns:
+        transformed_rotations: Array of shape (5,)
+            containing transformed rotation.
+    """
+    def __init__(self, num_pose_dims):
+        self.num_pose_dims = num_pose_dims
+        super(TransformRotation, self).__init__()
+
+    def call(self, rotations):
+        return transform_rotation(rotations, self.num_pose_dims)
+
+
+class ConcatenatePoses(Processor):
```

Comment:
Do we really need this as a processor? If it would not be extensivel used in other pipelines, we should maybe just call it as backend function in the call of the pipeline in which is being used.

Reply:
I don't really know. Just something to think about.

Reply:
`ConcatenatePoses` is used in the pipeline `AugmentEfficientPose` which is inherited from the class `SequentialProcessor`.  Hence the pipeline `AugmentEfficientPose` does not have call() function.